crates/core/component/distributions/src/component.rs

```diff
@@ -76,21 +81,66 @@ trait DistributionManager: StateWriteExt {
         tracing::debug!(
             number_of_blocks_in_epoch = num_blocks,
             staking_issuance_per_block,
-            "calculating issuance per epoch"
+            "calculating staking issuance per epoch"
         );
 
-        let new_issuance_for_epoch = staking_issuance_per_block
+        let new_staking_issuance_for_epoch = staking_issuance_per_block
             .checked_mul(num_blocks as u128) /* Safe to cast a `u64` to `u128` */
             .expect("infaillible unless issuance is pathological");
 
-        tracing::debug!(?new_issuance_for_epoch, "computed new issuance for epoch");
+        tracing::debug!(
+            ?new_staking_issuance_for_epoch,
+            "computed new staking issuance for epoch"
+        );
 
-        Ok(Amount::from(new_issuance_for_epoch))
+        Ok(Amount::from(new_staking_issuance_for_epoch))
     }
 
     /// Update the object store with the new issuance of staking tokens for this epoch.
-    async fn distribute(&mut self, new_issuance: Amount) {
-        self.set_staking_token_issuance_for_epoch(new_issuance)
+    async fn define_staking_budget(&mut self) -> Result<()> {
+        let new_issuance = self.compute_new_staking_issuance().await?;
+        tracing::debug!(?new_issuance, "computed new staking issuance for epoch");
+        Ok(self.set_staking_token_issuance_for_epoch(new_issuance))
+    }
+
+    /// Computes total LQT reward issuance for the epoch.
+    async fn compute_new_lqt_issuance(&self) -> Result<Amount> {
+        use penumbra_sdk_sct::component::clock::EpochRead;
+
+        let current_block_height = self.get_block_height().await?;
+        let current_epoch = self.get_current_epoch().await?;
+        let epoch_length = current_block_height
+            .checked_sub(current_epoch.start_height)
+            .unwrap_or_else(|| panic!("epoch start height is less than or equal to current block height (epoch_start={}, current_height={}", current_epoch.start_height, current_block_height));
```

Comment:
nit: the error message should say "start height is _greater_ than current block height"